### ISSUE TITLE
Bun is crashing when using elysia autoload

### ISSUE THREAD

**Issue by FireSquid6** [oven-sh/bun#12605]
### How can we reproduce the crash?

Minimum reproduction: https://github.com/firesquid6/elysia-autoload-bun-crash

I am doing `bun run server.ts` to run the app.

I'm trying to use [elysia autoload](https://github.com/kravetsone/elysia-autoload). The stack trace is from using Bun v1.1.17.

I figured the crash was just because I was on an old version of bun. I tried using bun v1.1.18 (unstable branch of nixpkgs), and I am getting the same error.

### Stack Trace (bun.report)

Bun v1.1.17 ([`bb66bba`](<https://github.com/oven-sh/bun/tree/bb66bba1bf6d57983e7efef7ab706bce12137028>)) on linux x86_64 [RunCommand]

Segmentation fault at address 0x0000001A

- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/ld-temp.o#L0>): `WTF::Unicode::convertReplacingInvalidSequences`
- `??`
- *1 unknown/js code*
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/ld-temp.o#L0>): `llint_relativePCBase`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/ld-temp.o#L0>): `JSC::Interpreter::executeCall`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/ld-temp.o#L0>): `JSC::JSModuleEnvironment::getOwnPropertySlot`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/ld-temp.o#L0>): `JSC::VM::drainMicrotasks`
- [`StringImpl.h:262`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028//build/bun/bun-webkit/include/wtf/text/StringImpl.h#L262>): `Bun::ProcessBindingUV::jsErrname`
- [`Strong.zig:118`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/bun.js/Strong.zig#L118>): `src.bun.js.api.Timer.TimerObject.runImmediateTask`
- [`interpreter.zig:7822`](<https://github.com/oven-sh/bun/blob/bb66bba1bf6d57983e7efef7ab706bce12137028/src/shell/interpreter.zig#L7822>): `src.bun.js.event_loop.EventLoop.tickQueueWithCount__anon_114135`

**Comment by FireSquid6** [comment]
hold on bun v1.1.20 is not out on nixpkgs so I gotta do some annoying manual stuff. I'll give another comment about that output.

**Comment by FireSquid6** [comment]
Yup. Crash still happening on v1.1.20. Here's the stack trace there:

Bun v1.1.20 ([`ae19489`](<https://github.com/oven-sh/bun/tree/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66>)) on linux x86_64 [AutoCommand]

Segmentation fault at address 0x0000001A

- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/ld-temp.o#L0>): `WTF::Unicode::convertReplacingInvalidSequences`
- `??`
- *1 unknown/js code*
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/ld-temp.o#L0>): `llint_relativePCBase`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/ld-temp.o#L0>): `JSC::Interpreter::executeCall`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/ld-temp.o#L0>): `JSC::JSModuleEnvironment::getOwnPropertySlot`
- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/ld-temp.o#L0>): `JSC::VM::drainMicrotasks`
- [`StringImpl.h:0`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66//build/bun/bun-webkit/include/wtf/text/StringImpl.h#L0>): `Bun::ProcessBindingUV::jsErrname`
- [`unbounded_queue.zig:129`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/src/bun.js/unbounded_queue.zig#L129>): `src.bun.js.module_loader.RuntimeTranspilerStore.drain`
- [`event_loop.zig:1211`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/src/bun.js/event_loop.zig#L1211>): `src.bun.js.event_loop.EventLoop.tickQueueWithCount__anon_117904`

**Comment by paperdave** [comment]
thanks for the reproduction. it happens instantly for me. i've seen `convertReplacingInvalidSequences` floating around in a lot of other stack traces but i didnt have a good repro.

**Comment by bili-jing** [comment]
I have the same problem. Can this be fixed with bun, or does it require elysia-autoload?

**Comment by paperdave** [comment]
yeah, this is a bug in bun

**Comment by FireSquid6** [comment]
There's currently some discussion on the elysia discord about this very issue. #12606 could possibly be related.

**Comment by bili-jing** [comment]
Elysia can be reduced to 1.0.27

**Comment by kravetsone** [comment]
HI! Im [elysia-autoload](https://github.com/kravetsone/elysia-autoload) maintainer

The most interesting thing is that everything works on Windows... but when i switched to WSL, i got the same error

**Comment by kravetsone** [comment]
i can fix it))

**Comment by FireSquid6** [comment]
I'm still getting an error (not bun crash though) when using autoload version 1.1.0. This one could be my bad though. Here's what I'm attempting to do:

https://github.com/FireSquid6/elysia-autoload-bun-crash/tree/kravetsone-fix

**Comment by kravetsone** [comment]
> Я все еще получаю ошибку (но не крах Bun) при использовании версии автозагрузки 1.1.0. Но это может быть моей ошибкой. Вот что я пытаюсь сделать:
> 
> https://github.com/FireSquid6/elysia-autoload-bun-crash/tree/kravetsone-fix
> 
> Хотя это обсуждение, возможно, следует перенести в репозиторий elysia-autoload, а не сюда, если только он не даст сбой.

latest elysia & autoload?

which error?

**Comment by paperdave** [comment]
the fact that this crashes bun is automatically a bug in bun that we will have to fix eventually, however it is nice to see a workaround in elysia.

**Comment by kravetsone** [comment]
> the fact that this crashes bun is automatically a bug in bun that we will have to fix eventually, however it is nice to see a workaround in elysia.

yeah i fix it but we need to find "why" it cause... i just export types with `export type * from` instead of `export * from` for two package and it solve the problem 

idk why it cause only with elysia@1.1 ....
